src/lgdo/lh5/store.py

```diff
@@ -0,0 +1,1425 @@
+            return self.files[lh5_file]
+        if self.base_path != "":
+            full_path = os.path.join(self.base_path, lh5_file)
+        else:
+            full_path = lh5_file
+        if mode != "r":
+            directory = os.path.dirname(full_path)
+            if directory != "" and not os.path.exists(directory):
+                log.debug(f"making path {directory}")
+                os.makedirs(directory)
+        if mode == "r" and not os.path.exists(full_path):
+            raise FileNotFoundError(f"file {full_path} not found")
+        if mode != "r" and os.path.exists(full_path):
+            log.debug(f"opening existing file {full_path} in mode '{mode}'")
+        h5f = h5py.File(full_path, mode)
+        if self.keep_open:
+            self.files[lh5_file] = h5f
+        return h5f
+
+    def gimme_group(
+        self,
+        group: str | h5py.Group,
+        base_group: h5py.Group,
+        grp_attrs: dict[str, Any] = None,
+        overwrite: bool = False,
+    ) -> h5py.Group:
+        """
+        Returns an existing :class:`h5py` group from a base group or creates a
+        new one. Can also set (or replace) group attributes.
+
+        Parameters
+        ----------
+        group
+            name of the HDF5 group.
+        base_group
+            HDF5 group to be used as a base.
+        grp_attrs
+            HDF5 group attributes.
+        overwrite
+            whether overwrite group attributes, ignored if `grp_attrs` is
+            ``None``.
+        """
+        if not isinstance(group, h5py.Group):
+            if group in base_group:
+                group = base_group[group]
+            else:
+                group = base_group.create_group(group)
+                if grp_attrs is not None:
+                    group.attrs.update(grp_attrs)
+                return group
+        if (
+            grp_attrs is not None
+            and len(set(grp_attrs.items()) ^ set(group.attrs.items())) > 0
+        ):
+            if not overwrite:
+                raise RuntimeError("grp_attrs != group.attrs but overwrite not set")
+            else:
+                log.debug(f"overwriting {group}.attrs...")
+                for key in group.attrs.keys():
+                    group.attrs.pop(key)
+                group.attrs.update(grp_attrs)
+        return group
+
+    def get_buffer(
+        self,
+        name: str,
+        lh5_file: str | h5py.File | list[str | h5py.File],
+        size: int = None,
+        field_mask: dict[str, bool] | list[str] | tuple[str] = None,
+    ) -> LGDO:
+        """Returns an LH5 object appropriate for use as a pre-allocated buffer
+        in a read loop. Sets size to `size` if object has a size.
+        """
+        obj, n_rows = self.read_object(name, lh5_file, n_rows=0, field_mask=field_mask)
+        if hasattr(obj, "resize") and size is not None:
+            obj.resize(new_size=size)
+        return obj
+
+    def read_object(
```

Comment:
I propose to rename `read_object()` and `write_object()` to just `read()` and `write()`. "object" sounds redundant to me.